subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java

```diff
@@ -411,14 +377,14 @@ public File getFile() {
         });
     }
 
-    private void configureBuild(Project project) {
+    private static void configureBuild(Project project) {
         project.getTasks().named(JavaBasePlugin.BUILD_NEEDED_TASK_NAME, task -> addDependsOnTaskInOtherProjects(task, true,
             JavaBasePlugin.BUILD_NEEDED_TASK_NAME, TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME));
         project.getTasks().named(JavaBasePlugin.BUILD_DEPENDENTS_TASK_NAME, task -> addDependsOnTaskInOtherProjects(task, false,
             JavaBasePlugin.BUILD_DEPENDENTS_TASK_NAME, TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME));
     }
 
-    private void configureConfigurations(Project project, JavaPluginExtension extension, SourceSet mainSourceSet) {
+    private void configureConfigurations(Project project, SourceSet mainSourceSet, PublishArtifact jarArtifact) {
```

Comment:
While you're refactoring the rest of this class - this name is abominable.  I know it fits the pattern of `configureXYZ()` methods here, but really, it sounds like a catch-all for doing anything not covered by the other, more sharply defined configure methods.  It doesn't really clearly indicate to me what it's actually doing - deprecating some legacy configurations and then setting up publishing and variants.  Perhaps it could be split into 2 methods with better names?

Reply:
This method is not deprecating legacy configurations. It is _creating_ the `apiElements` and `runtimeElements` configurations, then deprecating them for declaration. 

Though, I've renamed this method to `createConsumableConfigurations`, and moved the logic of creating and configuring the `apiElements` and `runtimeElements` into separate methods, `createApiElements` and `createRuntimeElements`. Now it is more clear that this method is creating configurations, and it offloads most of its logic to two separate methods which are each concerned with a single configuration.

Reply:
Great, thanks!